### ISSUE TITLE
Debug Error

### ISSUE THREAD

**Issue by dfhljf** [DonJayamanne/pythonVSCode#40]
when I start python debuger, got a error that print "Unexpected number. Stopping the debug adapter".
I reinstall vsc and this problem can't be solved.please help me.

**Comment by DonJayamanne** [comment]
Hi, when you encounter this error please could you open the developer tools and let me know if there is anything in the console window? If yes, please provide the details.

**Comment by lyyyuna** [comment]
@DonJayamanne Hi, I enounter the same error. Here is the log in the developer tools.
`Unexpected end of input. Stopping the debug adapter.
request 'setFunctionBreakpoints': unrecognized request: Error: request 'setFunctionBreakpoints': unrecognized request
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:78:182
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:17247)
    at Object.n.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20575)
    at n.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21896)
    at n.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21413)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:29436
    at t.dispatch (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:30522)
    at t.handleData (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:30032)
    at Socket.<anonymous> (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:29302)
    at emitOne (events.js:77:13)
WARNING: Promise with no error callback:12
Object {exception: null, error: Error: request 'setFunctionBreakpoints': unrecognized request
    at file:///C:/Program Files (x86)/…, promise: n.u…d.t.C…s.d…e.cancel, handler: undefined, id: 12…}error: Error: request 'setFunctionBreakpoints': unrecognized request
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:78:182
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:17247)
    at Object.n.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20575)
    at n.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21896)
    at n.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21413)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:29436
    at t.dispatch (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:30522)
    at t.handleData (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:30032)
    at Socket.<anonymous> (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:29302)
    at emitOne (events.js:77:13)exception: nullhandler: undefinedid: 12parent: undefinedpromise: n.undefined.t.Class.define.cancel__proto__: Object
`
My system is win10(Chinese)

**Comment by dbw9580** [comment]
Same problem with @lyyyuna :
FYI, I just managed to fix this somehow, which I believe was due to Chinese characters in the file path. After moving the file to another directory with all ASCII characters, the debugging functions fine. @lyyyuna 

Also, I find the "python.pythonPath" preference not working, because I have to expose python executable to PATH, otherwise when I start debugging, the debug console says "spawn python ENOENT" and the developer tools console gives:

>request 'setFunctionBreakpoints': unrecognized request: Error: request 'setFunctionBreakpoints': unrecognized request
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:78:182
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:17247)
    at Object.n.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20575)
    at n.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21896)
    at n.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21413)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:29436
    at t.dispatch (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:30522)
    at t.handleData (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:30032)
    at Socket.<anonymous> (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:77:29302)
    at emitOne (events.js:77:13)

**Comment by lyyyuna** [comment]
Hi, @dbw9580 , thanks.

But now when 
1. vscode in Chinses char directory (installed in 'c:\users\中文用户名\appdata' by default)
2. python.exe in ASCII char directory
3. file in ASCII char directory
The problem still exists.

**Comment by dbw9580** [comment]
@lyyyuna :
Then your situation might be different from mine. I suggest waiting for the author to look into this.

For your reference, what I have done that works for me is:
* expose python executable to system PATH:
* leave the extension's "python.pythonPath" unchanged, i.e. "python".

**Comment by DonJayamanne** [comment]
Hi, please could you install the latest version of VSCode and the latest updates and test this.

**Comment by wsg36236** [comment]
@DonJayamanne 

I have the same problem too, and my VS code version is 1.1.1.1

**Comment by liuce-** [comment]
@DonJayamanne 
I have the same problem too.
My vs code version is 1.1.1 
When i debug python code, it shows "Unexpected end of input. Stopping the debug adapter."

**Comment by leesoh** [comment]
Same issue here, here's the debug log.
```
Cannot read property 'getActiveEditor' of undefined: TypeError: Cannot read property 'getActiveEditor' of undefined
    at t.getActiveEditorInput (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:101:19916)
    at e.getActiveEditorInput (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:94:11666)
    at t.getFilePath (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:82:9120)
    at t.Object.defineProperty.get [as file] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:82:8609)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:4134
    at String.replace (native)
    at t.e.__resolveString (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:4103)
    at t.e.resolveAny (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:3923)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:103:7306
    at Array.forEach (native)
WARNING: Promise with no error callback:2
Object
TypeError: Cannot read property 'getActiveEditor' of undefined
    at t.getActiveEditorInput (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:101:19916)
    at e.getActiveEditorInput (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:94:11666)
    at t.getFilePath (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:82:9120)
    at t.Object.defineProperty.get [as file] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:82:8609)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:4134
    at String.replace (native)
    at t.e.__resolveString (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:4103)
    at t.e.resolveAny (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:3923)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:103:7306
    at Array.forEach (native)
Cannot convert undefined or null to object: TypeError: Cannot convert undefined or null to object
    at Function.e.adapt (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:69:7523)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:69:5916
    at Object.h [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:22735)
    at Object.__dirname.undefined.P.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:25897)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define._completed (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27164)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:31:18684
    at C:\Program Files (x86)\Microsoft VS Code\resources\app\node_modules\graceful-fs\graceful-fs.js:76:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:404:3)
WARNING: Promise with no error callback:5
Object
Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  40995  for  PowerShell.ExpandAlias
Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  40991  for  extension.align
[Extension Host] ---------------------------
[Extension Host] child_process.spawn in jediProxy
[Extension Host] Value of pythonSettings.pythonPath is :python
[Extension Host] ---------------------------
[Extension Host] child_process.spawn in jediProxy
[Extension Host] Value of pythonSettings.pythonPath is :python
[Extension Host] ---------------------------
[Extension Host] child_process.spawn in jediProxy
[Extension Host] Value of pythonSettings.pythonPath is :python
[Extension Host] ---------------------------
[Extension Host] child_process.spawn in jediProxy
[Extension Host] Value of pythonSettings.pythonPath is :python
[Extension Host] ---------------------------
[Extension Host] child_process.spawn in jediProxy
[Extension Host] Value of pythonSettings.pythonPath is :python
[Extension Host] ---------------------------
[Extension Host] child_process.spawn in jediProxy
[Extension Host] Value of pythonSettings.pythonPath is :python
[Extension Host] ---------------------------
[Extension Host] stderr jediProxy
[Extension Host] Error (stderr) Traceback (most recent call last):
  File "completion.py", line 313, in watch
    self._process_request(self._input.readline())
  File "completion.py", line 304, in _process_request
    request.get('prefix', '')))
  File "completion.py", line 136, in _serialize_completions
    completions = script.completions()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\api\__init__.py", line 192, in completions
    completion_names = get_completions(user_stmt, b)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\api\__init__.py", line 178, in get_completions
    completion_names += self._simple_complete(path, dot, like)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\api\__init__.py", line 254, in _simple_complete
    scopes = list(self._prepare_goto(path, True))
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\api\__init__.py", line 298, in _prepare_goto
    scopes = self._evaluator.eval_element(eval_stmt)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 169, in eval_element
    return self._eval_atom(element)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 230, in _eval_atom
    return self.find_types(scope, atom, stmt.start_pos, search_global=True)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 120, in find_types
    return f.find(scopes, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 87, in find
    types = self._names_to_types(names, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 233, in _names_to_types
    new_types = _name_to_types(self._evaluator, name, self.scope)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 278, in _name_to_types
    types += _remove_statements(evaluator, typ, name)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 323, in _remove_statements
    types += evaluator.eval_statement(stmt, seek_name=name)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 23, in run
    result = func(evaluator, stmt, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 135, in eval_statement
    types = self.eval_element(stmt.get_rhs())
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 187, in eval_element
    types = self.eval_trailer(types, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 259, in eval_trailer
    new_types += self.execute(typ, node, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 299, in execute
    types = func(self, arguments)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\_compatibility.py", line 124, in <lambda>
    return lambda *args, **kwargs: self.func(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 560, in py__call__
    return FunctionExecution(evaluator, self, params).get_return_types()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 100, in run
    result = func(execution, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 624, in get_return_types
    types += self._evaluator.eval_element(r.children[1])
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 187, in eval_element
    types = self.eval_trailer(types, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 259, in eval_trailer
    new_types += self.execute(typ, node, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 299, in execute
    types = func(self, arguments)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\_compatibility.py", line 124, in <lambda>
    return lambda *args, **kwargs: self.func(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 560, in py__call__
    return FunctionExecution(evaluator, self, params).get_return_types()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 100, in run
    result = func(execution, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 624, in get_return_types
    types += self._evaluator.eval_element(r.children[1])
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 187, in eval_element
    types = self.eval_trailer(types, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 259, in eval_trailer
    new_types += self.execute(typ, node, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 299, in execute
    types = func(self, arguments)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 364, in py__call__
    return Function.py__call__(self, evaluator, params)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\_compatibility.py", line 122, in <lambda>
    return lambda *args, **kwargs: self.func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 560, in py__call__
    return FunctionExecution(evaluator, self, params).get_return_types()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 100, in run
    result = func(execution, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 624, in get_return_types
    types += self._evaluator.eval_element(r.children[1])
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 169, in eval_element
    return self._eval_atom(element)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 230, in _eval_atom
    return self.find_types(scope, atom, stmt.start_pos, search_global=True)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 120, in find_types
    return f.find(scopes, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 87, in find
    types = self._names_to_types(names, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 233, in _names_to_types
    new_types = _name_to_types(self._evaluator, name, self.scope)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 278, in _name_to_types
    types += _remove_statements(evaluator, typ, name)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 323, in _remove_statements
    types += evaluator.eval_statement(stmt, seek_name=name)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 23, in run
    result = func(evaluator, stmt, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 135, in eval_statement
    types = self.eval_element(stmt.get_rhs())
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 187, in eval_element
    types = self.eval_trailer(types, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 259, in eval_trailer
    new_types += self.execute(typ, node, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 299, in execute
    types = func(self, arguments)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 364, in py__call__
    return Function.py__call__(self, evaluator, params)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\_compatibility.py", line 122, in <lambda>
    return lambda *args, **kwargs: self.func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 560, in py__call__
    return FunctionExecution(evaluator, self, params).get_return_types()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 100, in run
    result = func(execution, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 624, in get_return_types
    types += self._evaluator.eval_element(r.children[1])
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 169, in eval_element
    return self._eval_atom(element)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 230, in _eval_atom
    return self.find_types(scope, atom, stmt.start_pos, search_global=True)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 120, in find_types
    return f.find(scopes, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 87, in find
    types = self._names_to_types(names, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 233, in _names_to_types
    new_types = _name_to_types(self._evaluator, name, self.scope)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 278, in _name_to_types
    types += _remove_statements(evaluator, typ, name)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 323, in _remove_statements
    types += evaluator.eval_statement(stmt, seek_name=name)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 23, in run
    result = func(evaluator, stmt, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 135, in eval_statement
    types = self.eval_element(stmt.get_rhs())
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 187, in eval_element
    types = self.eval_trailer(types, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 259, in eval_trailer
    new_types += self.execute(typ, node, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 299, in execute
    types = func(self, arguments)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 364, in py__call__
    return Function.py__call__(self, evaluator, params)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\_compatibility.py", line 122, in <lambda>
    return lambda *args, **kwargs: self.func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 560, in py__call__
    return FunctionExecution(evaluator, self, params).get_return_types()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\recursion.py", line 100, in run
    result = func(execution, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\representation.py", line 624, in get_return_types
    types += self._evaluator.eval_element(r.children[1])
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 187, in eval_element
    types = self.eval_trailer(types, trailer)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 267, in eval_trailer
    new_types += get(self, node)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\iterable.py", line 229, in get_index_types
    types += self.get_exact_index_types(index.obj)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\iterable.py", line 255, in get_exact_index_types
    return self._evaluator.eval_element(self._items()[mixed_index])
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 169, in eval_element
    return self._eval_atom(element)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 230, in _eval_atom
    return self.find_types(scope, atom, stmt.start_pos, search_global=True)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\__init__.py", line 120, in find_types
    return f.find(scopes, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\debug.py", line 52, in wrapper
    result = func(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 87, in find
    types = self._names_to_types(names, search_global)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 233, in _names_to_types
    new_types = _name_to_types(self._evaluator, name, self.scope)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\cache.py", line 41, in wrapper
    rv = function(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\finder.py", line 269, in _name_to_types
    for_types = iterable.get_iterator_types(for_types)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\iterable.py", line 418, in get_iterator_types
    result += it.iter_content()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\iterable.py", line 575, in iter_content
    module = self.var_args.get_parent_until()
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\evaluate\param.py", line 56, in get_parent_until
    return element.get_parent_until(*args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\_compatibility.py", line 124, in <lambda>
    return lambda *args, **kwargs: self.func(obj, *args, **kwargs)
  File "C:\Users\e591056\.vscode\extensions\donjayamanne.python-0.3.9\pythonFiles\jedi\parser\tree.py", line 123, in get_parent_until
    while scope.parent is not None:
AttributeError: 'ListComprehension' object has no attribute 'parent'


Debug adapter process has terminated unexpectedly
Canceled: Error: Canceled
    at Object.w [as create] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:30:28776)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:103:26411
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27240)
    at n (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:24945)
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
WARNING: Promise with no error callback:undefined
Error: Promise did not implement oncancel(…)
Debug adapter process has terminated unexpectedly
Canceled: Error: Canceled
    at Object.w [as create] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:30:28776)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:103:26411
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27240)
    at n (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:24945)
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
Canceled: Error: Canceled
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:79:17254
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26888)
    at __dirname.undefined.t.Class.derive._cancelAction (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27979)
    at Object.__dirname.undefined.R.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:25416)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26888)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:79:19337
WARNING: Promise with no error callback:90
Object
WARNING: Promise with no error callback:undefined
Error: Promise did not implement oncancel(…)
Debug adapter process has terminated unexpectedly
Canceled: Error: Canceled
    at Object.w [as create] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:30:28776)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:103:26411
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27240)
    at n (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:24945)
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
Canceled: Error: Canceled
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:79:17254
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26888)
    at __dirname.undefined.t.Class.derive._cancelAction (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27979)
    at Object.__dirname.undefined.R.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:25416)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26888)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:79:19337
WARNING: Promise with no error callback:93
Object {exception: null, error: Array[1], promise: _…e.u…d.t.C…s.d…e._oncancel, handler: undefined, id: 93…}error: Array[1]exception: nullhandler: undefinedid: 93parent: undefinedpromise: __dirname.undefined.t.Class.derive._oncancel_errorId: 93_isException: false_listeners: null_nextState: null_oncancel: null_state: Object_completed: i()arguments: (...)caller: (...)length: 0name: "i"prototype: Object__proto__: ()<function scope>_error: i()_notify: p(e,t)_progress: i()_setCompleteValue: i()_setErrorValue: i()cancel: i()done: nullenter: (e)name: "error"then: null__proto__: Object_value: Array[1]0: Error: Canceled
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:79:17254
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:23074)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26402)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26888)
    at __dirname.undefined.t.Class.derive._cancelAction (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27979)
    at Object.__dirname.undefined.R.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:25416)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:27723)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:26888)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:79:19337length: 1__proto__: Array[0]done: (e,t)arguments: (...)caller: (...)length: 2name: ""prototype: Object__proto__: ()<function scope>then: (e,t)arguments: (...)caller: (...)length: 2name: ""prototype: Object__proto__: ()<function scope>__proto__: t_cancelAction: ()_cleanupAction: ()_oncancel: nullconstructor: (e,n)__proto__: Object__proto__: Object__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Object()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()propertyIsEnumerable: propertyIsEnumerable()toLocaleString: toLocaleString()toString: toString()valueOf: valueOf()get __proto__: get __proto__()set __proto__: set __proto__()
WARNING: Promise with no error callback:undefined
Error: Promise did not implement oncancel(…)

```

**Comment by DonJayamanne** [comment]
Thanks

**Comment by leesoh** [comment]
No problem. Let me know if you need anything else for either this one or #111

**Comment by DonJayamanne** [comment]
@leesoh , please could you 
1.confirm you do not have any unicode characters in your path. 
2. dump the contents of your launch.json file here a
3. let me know if you are using a virtual enviroment
4. finally please could you try debugging something as simple as the following code and let me know how that goes. Thanks
```python
one = 1 + 2
print one
```

FYI - I'll be closing #40, as that can be considered a duplicate of this issue.

**Comment by leesoh** [comment]
1. None
2. 
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python",
            "type": "python",
            "request": "launch",
            "stopOnEntry": true,
            "program": "${file}",
            "pythonPath": "/usr/bin/python",
            "debugOptions": [
                "WaitOnAbnormalExit",
                "WaitOnNormalExit",
                "RedirectOutput"
            ]
        },
        {
            "name": "Python Console App",
            "type": "python",
            "request": "launch",
            "stopOnEntry": true,
            "program": "${file}",
            "externalConsole": true,
            "debugOptions": [
                "WaitOnAbnormalExit",
                "WaitOnNormalExit"
            ]
        },
        {
            "name": "Django",
            "type": "python",
            "request": "launch",
            "stopOnEntry": true,
            "program": "${workspaceRoot}/manage.py",
            "args": [
                "runserver",
                "--noreload"
            ],
            "debugOptions": [
                "WaitOnAbnormalExit",
                "WaitOnNormalExit",
                "RedirectOutput",
                "DjangoDebugging"
            ]
        },
        {
            "name": "Watson",
            "type": "python",
            "request": "launch",
            "stopOnEntry": true,
            "program": "${workspaceRoot}/console.py",
            "args": [
                "dev",
                "runserver",
                "--noreload=True"
            ],
            "debugOptions": [
                "WaitOnAbnormalExit",
                "WaitOnNormalExit",
                "RedirectOutput"
            ]
        }
    ]
}
```
3. VMware Workstation
4. I was debugging with `print "hello world"`

**Comment by DonJayamanne** [comment]
Hi @leesoh , please could you retest this and let me know how it goes.
By the way, please ensure you have cleared your watch window before starting the debug session.
Thanks

**Comment by leesoh** [comment]
Hi @DonJayamanne, thanks for the update but sadly I'm having the same problem.

**Comment by DonJayamanne** [comment]
Hi @leesoh , please could you open the developer tools and get the errors from the console window and paste them here. Thanks for your help, I'd like to get to the bottom of this.

**Comment by leesoh** [comment]
```
workbench.main.js:87 Debug adapter process has terminated unexpectedlye.doShow @ workbench.main.js:87e.show @ workbench.main.js:87t.onServerExit @ workbench.main.js:83(anonymous function) @ workbench.main.js:83emitTwo @ events.js:105emit @ events.js:185ChildProcess._handle.onexit @ internal/child_process.js:204
workbench.main.js:87 Canceled: Error: Canceled
    at Object.E [as create] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:31:17647)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:104:23592
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:16345)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:19673)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20994)
    at __dirname.undefined.t.Class.define._error (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20511)
    at n (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:18216)
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:16345)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:19673)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20994)e.doShow @ workbench.main.js:87e.show @ workbench.main.js:87(anonymous function) @ workbench.main.js:88__dirname.undefined.t.Class.define.done @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23__dirname.undefined.t.Class.define.then @ workbench.main.js:23__dirname.undefined.O.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._setCompleteValue @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23__dirname.undefined.t.Class.define.then @ workbench.main.js:23__dirname.undefined.O.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._setCompleteValue @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._error @ workbench.main.js:23n @ workbench.main.js:23__dirname.undefined.t.Class.define.then @ workbench.main.js:23__dirname.undefined.O.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define._setCompleteValue @ workbench.main.js:23p @ workbench.main.js:23__dirname.undefined.F.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define.cancel @ workbench.main.js:23__dirname.undefined.t.Class.derive._cancelAction @ workbench.main.js:23__dirname.undefined.R.enter @ workbench.main.js:23__dirname.undefined.t.Class.define._run @ workbench.main.js:23__dirname.undefined.t.Class.define.cancel @ workbench.main.js:23(anonymous function) @ workbench.main.js:83(anonymous function) @ workbench.main.js:83
workbench.main.js:118 Canceled: Error: Canceled
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:83:55
    at Object.p [as _notify] (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:16345)
    at Object.__dirname.undefined.F.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:19673)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20994)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20159)
    at __dirname.undefined.t.Class.derive._cancelAction (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:21250)
    at Object.__dirname.undefined.R.enter (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:18687)
    at __dirname.undefined.t.Class.define._run (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20994)
    at __dirname.undefined.t.Class.define.cancel (file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:23:20159)
    at file:///C:/Program Files (x86)/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:83:1623e.onUnexpectedError @ workbench.main.js:118(anonymous function) @ workbench.main.js:118e.onUnexpectedError @ workbench.main.js:31u @ workbench.main.js:31(anonymous function) @ workbench.main.js:32(anonymous function) @ workbench.main.js:32
workbench.main.js:32 WARNING: Promise with no error callback:9
workbench.main.js:32 Object {exception: null, error: Array[1], promise: _…e.u…d.t.C…s.d…e._oncancel, handler: undefined, id: 9…}
workbench.main.js:32 WARNING: Promise with no error callback:undefined
workbench.main.js:32 Error: Promise did not implement oncancel(…)
```

**Comment by DonJayamanne** [comment]
Hi @leesoh , I'll close this issue as it seems to be a duplicate of #55
